src/main/java/net/barrage/school/java/ecatalog/app/XmlPS.java

```diff
@@ -0,0 +1,73 @@
+package net.barrage.school.java.ecatalog.app;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.dataformat.xml.XmlMapper;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import net.barrage.school.java.ecatalog.config.ProductSourceProperties;
+import net.barrage.school.java.ecatalog.model.Product;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+@RequiredArgsConstructor
+public class XmlPS implements ProductSource {
+    private static final Logger log = LoggerFactory.getLogger(XmlPS.class);
+    private final XmlMapper xmlMapper;
+    private final ProductSourceProperties.SourceProperty property;
+
+    @SneakyThrows
+    @Override
+    public List<Product> getProducts() {
+        try {
```

Comment:
U don't need `SneakyThrows` if u have try-block, or u don't need try-block if u use  `SneakyThrows`. Pls leave one, and remove another)